### PR TITLE
[WIP] Include 'digest' in release stream JSON

### PR DESCRIPTION
**Pull request by tbielawa**
Automating release signing will be simpler if the release stream
includes the digest information as well.

Work in progress. I'm not sure what to do about `digest, _ := c.digestForArtifacts(latest.Name)`

I could use some help with that part.

### CODE REVIEW

**Comment by tbielawa** [comment]
Also it would be cool if we could include the canonical client download location in there

`clientsDownloadURL: https://mirror.openshift.com/pub/openshift-v4/clients/ocp/${NAME}`

My initial thought on that, if we change that location in the future then we'd have to update the code as well. Thoughts on that?
Also `clientsDownloadURL` is specific. We also include the installers in there. I'm open to a better name, and not offended if you shoot this idea down entirely @smarterclayton

File: cmd/release-controller/http.go [hunk 1]
```diff
@@ -280,10 +280,12 @@ func (c *Controller) apiReleaseLatest(w http.ResponseWriter, req *http.Request)
 	}
 
 	downloadURL, _ := c.urlForArtifacts(latest.Name)
+	digest, _ := c.digestForArtifacts(latest.Name)
```
Comment (tbielawa) [inline cmd/release-controller/http.go]: This is obviously not implemented.

Comment (stevekuznetsov) [reply cmd/release-controller/http.go]: The `locateLatest` function returns a `TagReference`. That object has a `From` field which holds information like the `Name` and `Namespace` of the `ImageStreamTag`, `ImageStreamImage` or `DockerImage` that it is referencing. If you use the client to fetch the requisite item, the digest is:

 - `ImageStreamTag.Image.ObjectMeta.Name`
 - `ImageStreamImage.Image.ObjectMeta.Name`

In the case of a `DockerImage` you would need to parse out the `Name` field from the reference itself. I am not sure we need to handle this case.

**Pull request closed without merging**